climakitae/util/utils.py

```diff
@@ -772,90 +772,102 @@ def summary_table(data: xr.Dataset) -> pd.DataFrame:
+                )
+                return data
+    else:
+        data_type = data.attrs["data_type"]
 
     # Default lat/lon values in case other methods fail
     lat = None
     lon = None
 
     # Get latitude/longitude information
-    if selections.data_type == "Stations":
-        # Read stations database
-        stations_df = read_csv_file(stations_csv_path)
-        stations_df = stations_df.drop(columns=["Unnamed: 0"])
-
-        # Filter by selected station(s) - assume first station if multiple
-        selected_station = selections.stations[0]
-        station_data = stations_df[stations_df["station"] == selected_station]
-        lat = station_data["LAT_Y"].values[0]
-        lon = station_data["LON_X"].values[0]
-
-    elif selections.area_average == "Yes":
-        # For area average, use the mean lat/lon
-        lat = np.mean(selections.latitude)
-        lon = np.mean(selections.longitude)
-
-    elif selections.data_type == "Gridded" and selections.area_subset == "lat/lon":
-        # Finding avg. lat/lon coordinates from all grid-cells
-        lat = data.lat.mean().item()
-        lon = data.lon.mean().item()
-
-    elif selections.data_type == "Gridded" and selections.area_subset != "none":
-        # Find the avg. lat/lon coordinates from entire geometry within an area subset
-        boundaries = selections._geographies
-
-        # Making mapping for different geographies to different polygons
-        mapping = {
-            "CA counties": (
-                boundaries._ca_counties,
-                boundaries._get_ca_counties(),
-            ),
-            "CA Electric Balancing Authority Areas": (
-                boundaries._ca_electric_balancing_areas,
-                boundaries._get_electric_balancing_areas(),
-            ),
-            "CA Electricity Demand Forecast Zones": (
-                boundaries._ca_forecast_zones,
-                boundaries._get_forecast_zones(),
-            ),
-            "CA Electric Load Serving Entities (IOU & POU)": (
-                boundaries._ca_utilities,
-                boundaries._get_ious_pous(),
-            ),
-            "CA watersheds": (
-                boundaries._ca_watersheds,
-                boundaries._get_ca_watersheds(),
-            ),
-        }
-
-        # Finding the center point of the gridded WRF area
-        center_pt = (
-            mapping[selections.area_subset][0]
-            .loc[mapping[selections.area_subset][1][selections.cached_area[0]]]
-            .geometry.centroid
-        )
-        lat = center_pt.y
-        lon = center_pt.x
+    match data_type:
+        case "Stations":
+            # Read stations database
+            stations_df = read_csv_file(stations_csv_path)
+            stations_df = stations_df.drop(columns=["Unnamed: 0"])
+
+            # Filter by selected station(s) - assume first station if multiple
+            match type(data):
```

Comment:
typical use of match statements for type checking look like:
```python
match data:
    case xr.DataArray():
        # handle data array case
    case xr.Dataset():
        # handle dataset case
    case _:
        # raise an error because this is the wrong type
```